README.md

```diff
@@ -1,5 +1,65 @@
 # Go OpenSSL bindings for FIPS compliance
 
-This repository holds Go bindings into the OpenSSL library for FIPS compliance.
+[![Go Reference](https://pkg.go.dev/badge/github.com/golang-fips/openssl.svg)](https://pkg.go.dev/github.com/golang-fips/openssl)
 
-More information to be provided soon.
\ No newline at end of file
+The `openssl` package implements Go crypto primitives using OpenSSL shared libraries and cgo. When configured correctly, OpenSSL can be executed in FIPS mode, making the `openssl` package FIPS compliant.
+
+The `openssl` package is designed to be used as a drop-in replacement for the [boring](https://pkg.go.dev/crypto/internal/boring) package in order to facilitate integrating `openssl` inside a forked Go toolchain.
+
+## Disclaimer
+
+A program directly or indirectly using this package in FIPS mode can claim it is using a FIPS-certified cryptographic module (OpenSSL), but it can't claim the program as a whole is FIPS certified without passing the certification process, nor claim it is FIPS compliant without ensuring all crypto APIs and workflows are implemented in a FIPS-compliant manner.
+
+## Background
+
+FIPS 140-2 is a U.S. government computer security standard used to approve cryptographic modules. FIPS compliance may come up when working with U.S. government and other regulated industries.
+
+### Go FIPS compliance
+
+The Go `crypto` package is not FIPS certified, and the Go team has stated that it won't be, e.g. in [golang/go/issues/21734](https://github.com/golang/go/issues/21734#issuecomment-326980213) Adam Langley says:
+
+> The status of FIPS 140 for Go itself remains "no plans, basically zero chance".
+
+On the other hand, Google maintains a branch that uses cgo and BoringSSL to implement various crypto primitives: https://github.com/golang/go/blob/dev.boringcrypto/README.boringcrypto.md. As BoringSSL is FIPS 140-2 certified, an application using that branch is more likely to be FIPS 140-2 compliant, yet Google does not provide any liability about the suitability of this code in relation to the FIPS 140-2 standard.
+
+## Features
+
+### Multiple OpenSSL versions supported
+
+The `openssl` package has support for multiple OpenSSL versions, namely 1.0.2, 1.1.0, 1.1.1 and 3.0.x.
+
+All supported OpenSSL versions pass a small set of automatic tests that ensure they can be built and that there are no major regressions.
+These tests do not validate the cryptographic correctness of the `openssl` package.
+
+On top of that, the [Microsoft Go fork](https://github.com/microsoft/go) and the [RedHat Go fork](https://github.com/golang-fips/go) build and test a subset of the supported OpenSSL versions when integrated with the Go `crypto` package.
```

Comment:
> RedHat

I believe Red Hat should have a space, or RHEL would refer to the distro. (There's another instance in the `Acknowledgements` section too.)

But I'm not totally sure about calling it Red Hat's fork here, maybe "the golang-fips Go fork". And/or the repo could be updated to mention who's currently maintaining/using it.

Not to say I object to calling it Red Hat's, just want to point this out and IMO it's up to Red Hat folks. 😄

Reply:
Good point. I've renamed RedHat to Red Hat and changed the wording to make it clear that golang-fips/go is not Red Hat-specific but it's mainly maintained by them.